### PR TITLE
Karayieb code review

### PR DESCRIPTION
**Pull request by karayigit-emine**
The project was on time and it compiled successfully. I didn't have any issues. I added few things.

Here is the 3 changes that I made:

1) I changed quicktype as CrimeSpace for clarification in the code because quick type isn't very explanatory. Same as for QuickTypeParks. Credit: Prof. Jones videos

2)They didn't have validation for  crime data and I added this one. They also need validation for park data but I didn't add that but instead as a third change; Credit: Prof. Jones videos and example code 

3) I created parkSchema. json because they only had crime schema. I thought that would help them to create validation for their crime data.  Credit: https://app.quicktype.io/

Learnings:

I think that was a good learning experience for me because I was able to review my own project and able to see some mistakes. Also, i have learned validation and was able to practice creating json schema. 

Overall I think project is very good. I enjoyed reviewing it.  Hopefully, I added some value to the project.

### CODE REVIEW

File: IS7024Project/Pages/Index.cshtml.cs [hunk 1]
```diff
@@ -30,15 +30,35 @@ public void OnGet()
...
+               
+
+
 
 
                 //Consuming Crime data
                 string jsonString = webClient.DownloadString("https://raw.githubusercontent.com/JMFrank215/IS7024Project/master/PDI_Crime_Data.txt");
-                var crime = Crime.FromJson(jsonString);
-                ViewData["Crime"] = crime;
+
```
Comment (karayigit-emine) [inline IS7024Project/Pages/Index.cshtml.cs]: Added validation for received data for crime

File: IS7024Project/Crime.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@
 //
 //    var crime = Crime.FromJson(jsonString);
 
-namespace QuickType
+namespace CrimeSpace
```
Comment (karayigit-emine) [inline IS7024Project/Crime.cs:10]: Changed the Quicktype as CrimeSpace to be more descriptive.

File: IS7024Project/ParkSchema.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
+﻿{
```
Comment (karayigit-emine) [inline IS7024Project/ParkSchema.json:1]: Created parkschema.json for validation for future use.

File: IS7024Project/Crime.cs (hunk 2 above)
Comment (discospiff) [reply IS7024Project/Crime.cs:10]: A better namespace, yes.
You can specify the namespace in QuickType as well... I figured this out after I recorded the video.  :)

File: IS7024Project/Pages/Index.cshtml.cs [hunk 4]
```diff
@@ -30,15 +30,35 @@ public void OnGet()
                 string parksJSON = webClient.DownloadString("https://raw.githubusercontent.com/JMFrank215/IS7024Project/master/Parks_Data.txt");
                 var park = Parks.FromJson(parksJSON);
                 ViewData["Parks"] = park;
-                QuickTypeParks.Parks[] Parksparks = QuickTypeParks.Parks.FromJson(parksJSON);
+                ParkSpace.Parks[] Parksparks = ParkSpace.Parks.FromJson(parksJSON);
+                
+               
+               
+
+
```
Comment (discospiff) [inline IS7024Project/Pages/Index.cshtml.cs:38]: Probably don't need this much whitespace.  :)

File: IS7024Project/Pages/Index.cshtml.cs [hunk 5]
```diff
@@ -30,15 +30,35 @@ public void OnGet()
...
+               
+                JArray cjsonObject = JArray.Parse(jsonString);
+
+                if (cjsonObject.IsValid(cschema))
+                {
+                    var crime = Crime.FromJson(jsonString);
+                    ViewData["Crime"] = crime;
+                    
+                }
+
```
Comment (discospiff) [inline IS7024Project/Pages/Index.cshtml.cs:57]: Wow.  Cool.  But how are errors caught/handled?  :)

**Comment by discospiff** [comment]
Several good suggestions in this branch.
I recommend that the group review the changes carefully, and then consider merging.

**Review by JMFrank215 — APPROVED** [review]

**Pull request merged**